lisa/analysis/tasks.py

```diff
@@ -423,6 +423,46 @@ def df_task_total_residency(self, task):
 
         return residency_df
 
+    @df_task_total_residency.used_events
+    def df_tasks_total_residency(self, tasks=None, ascending=False, count=None):
+        """
+        DataFrame of tasks execution time on each CPU
+
+        :param tasks: List of tasks to report, all trace tasks by default
+        :type tasks: list(int or str)
+
+        :param ascending: Set True to plot before short running tasks,
+                          False by deafult
+        :type ascending: bool
+
+        :param count: Maximum number of tasks to report
+        :type count: int
+        """
+        if tasks is None:
+            tasks = list(self.trace.get_tasks().keys())
+        res_df = pd.DataFrame()
+
+        # Ensure we have a list of PIDs
+        pids = []
+        for task in tasks:
+            if isinstance(task, str):
+                pids.extend(self.trace.get_task_by_name(task))
+                continue
+            pids.extend([task])
+
+        for pid in pids:
+            task = self.trace.get_task_by_pid(pid)
```

Comment:
I think this still can return wrong values. I've seen cases where a pid had more than one comm value associated with it on some traces. AFAIU JobScheduler framework will create a pool of threads and reuse them to execute different functions and when it does it renames the task to something else.

So there's a chance `get_task_by_pid()` will return a task that is not what you intended to look at. At least you'll see the name and notice the wrong difference in this case. If someone supplied the pid they probably assume it'll print a specific task name.

Fixing and dealing with this might be outside the scope of this patch though...

Reply:
Interesting, maybe `get_task_by_pid()` should warn when it finds more than one comm for the same PID ?

Reply:
> Interesting, maybe `get_task_by_pid()` should warn when it finds more than one comm for the same PID ? Actually, it would even make sense to raise an exception if nothing sensible can be done.

`get_task_pid` does, but `get_task_by_pid` doesn't...

Reply:
Do the the 2 different functions make sense? Shall we keep get_task_pid() since it's a superset of get_task_by_pid()?

Reply:
Right, we should probably get rid of `get_task_by_pid` & `get_task_by_name`

Reply:
That's I fix for a different PR I guess... is there any other major concern about this PR?

Reply:
None from my side as long as we don't forget to deal with the issues raised here

Reply:
Maybe it's worth to open an issue to reference this discussion?

Reply:
Issue ticket #926 created to keep this discussion going.